### ISSUE TITLE
feature request: remove `flag` registrations from `sqlparser` or non-main packages?

### ISSUE THREAD

**Issue by dt** [vitessio/vitess#4089]
#### Use Case(s)

I'm interested in using (and contributing to) the parser and AST for use in a different project parsing MySQL-dialect SQL, but the `flag` registration side-effects of linking the current package makes that more difficult.

#### Feature Description

For packages that offer reusable functionality as standalone libraries, like `sqlparser`, avoid including runtime side-effects like`flag` registrations, preferring to move those to `main` packages that e.g. mutate package vars, or are passed as options, instead.

I obviously realize that this doesn't deliver any direct value to the project itself, so I of course understand if it isn't an appealing change, but on the off chance it sounds good, I'd be happy to send a PR.

**Comment by sougou** [comment]
Tagging @demmer. The only flags I see are in truncate_query.go.

I think this makes sense in principle. However, moving the flags to the main program may not be a good strategy because it removes the proximity of variables from where they're used.

Instead, you could implement a `RegisterFlags` in sqlparser that the main program has to call. And those  flag variables can be globals with defaults (in case main doesn't call `RegisterFlags`).

This is also not a very good strategy, but it's better than moving the flags into main.

I looked at moving the Truncate functions out of parser, but other functions in the parser use it. So, it's not worth going that route.

In other words, `RegisterFlags` seems like the best of the bad options.

**Comment by dt** [comment]
On further investigation, even if I wrapped these with a register helper, there are still flags pulled in via the dep on the `glog`, so I'm now thinking it probably just makes more sense for me to maintain a patch on my own fork, than rather add the cruft of the options described here.

**Comment by sougou** [comment]
We've discussed making sqlparser a separate and standalone project. I think we'll eventually get there. It may be too premature to do it right now, but it should be achievable once the project matures a bit.

Same thing for the mysql protocol.

**Comment by sougou** [comment]
I'll close this for now.
